airbyte-commons-worker/src/main/java/io/airbyte/workers/general/DefaultCheckConnectionWorker.java

```diff
@@ -78,28 +79,29 @@ public ConnectorJobOutput run(final StandardCheckConnectionInput input, final Pa
         WorkerUtils.gentleClose(process, 1, TimeUnit.MINUTES);
       }
 
+      final Optional<FailureReason> failureReason = WorkerUtils.getJobFailureReasonFromMessages(OutputType.CHECK_CONNECTION, messagesByType);
       final int exitCode = process.exitValue();
+      LOGGER.debug("Check connection job subprocess finished with exit code {}", exitCode);
+
       final Optional<AirbyteConnectionStatus> status = messagesByType
           .getOrDefault(Type.CONNECTION_STATUS, new ArrayList<>()).stream()
           .map(AirbyteMessage::getConnectionStatus)
           .findFirst();
 
-      if (status.isPresent() && exitCode == 0) {
+      final String defaultErrorMessage = String.format("Error checking connection, status: %s, exit code: %d", status, exitCode);
+
+      if (status.isEmpty()) {
+        LOGGER.error(defaultErrorMessage);
+        return WorkerUtils.getJobOutput(OutputType.CHECK_CONNECTION, failureReason, defaultErrorMessage, true);
```

Comment:
In reading this method being used it's not clear that this might actually throw - I would suggest either throwing here or having a different method that throws rather than having a boolean param.